eclib/JModel_m4.ec

```diff
@@ -250,6 +250,50 @@ op SMMULR (x y: W32.t) : W32.t =
   W32.of_int (IntDiv.(%/) (to_sint x * to_sint y + 2 ^ 31) (2 ^ 32)).
 op SMMULRcc x y g o = if g then SMMULR x y else o.
 
+abbrev get_hw (is_hi: bool) (x: W32.t) : W16.t =
```

Comment:
I know we have discussed this, and maybe I confuse you.
I think smul_hw should not be an abbrev because it is the generic definition. 
I think also that you can define the generic definition
op smul_hwcc hwx hwy x y g o = 
   if g then smul_hw hwx hwy x y else o.
op smula_hw hwx hwy x y acc = smul_hw x y + acc.
op smula_hwcc hwx hwy x y acc g o = if g then smula_hw x y else o.
then all definitions become simpler.

Reply:
I've just changed this, I did not use `smul_hw` to defined `smla_hw` to be consistent with the reference and Coq code (addition for Z instead of words).